### PR TITLE
evaluating all/any/allM/anyM on very simple lists

### PR DESCRIPTION
**Pull request by jvoigtlaender**
The actual motivation for drawing those up was that I saw this in code:
```haskell
 if all null [unreachedState] then ...
```
Wasn't sure whether that was really what the student intended or there was some logical error.

Had they been prodded to rewrite this to:
```haskell
 if null unreachedState then ...
```
it would have probably been clearer already at their end (and possibly been changed in case of a logical error).

The other proposed rules are all similar, more or less just applying the same idea to similar functions.

I also thought about further special casing for two-element-lists, like
```yaml
    - warn: {lhs: "all f [a,b]", rhs: f a && f b, name: Evaluate}
```
but I guess this could be undesirable sometimes (such as when `f` is a long expression). So I didn't include those.

### CODE REVIEW

File: data/hlint.yaml
Comment (yairchu) [inline data/hlint.yaml:969]: Should probably suggest `pure` rather than `return`

**Review by yairchu — CHANGES_REQUESTED** [review]
Note: Just my 2 cents (I'm not the maintainer so am not sure about the etiquette of reviewing)

**Comment by ndmitchell** [comment]
I'll make the change to pure after merging

**Pull request merged**
